File: leetcode/001-two-sum/main.rb
# @param {Integer[]} nums
# @param {Integer} target
# @return {Integer[]}
def two_sum(nums, target)
  for i in 0..nums.length
    for j in i+1..(nums.length-1)
      if (nums[i] + nums[j]) == target
        return [i, j]
      end
    end
  end
  return []
end

puts "001-two-sum"
nums = [2,7,11,15]
target = 9
puts two_sum(nums,target)

File: leetcode/001-two-sum/main.ts
function twoSum(nums: number[], target: number): number[] {
	let res: number[] = [];
	for(let i = 0; i < nums.length - 1; i++) {
		for(let j = i + 1; j < nums.length; j++) {
			if(nums[i] + nums[j] === target) {
				res[0] = i;
				res[1] = j;
				return res;
			}
		}
	}
	return res;
};

let nums: number[] = [2,7,11,15];
let target: number = 9;
let test: number[] = twoSum(nums, target);
console.log(test)